.github/workflows/lint.yml

```diff
@@ -0,0 +1,58 @@
+name: Mypy static type checking
```

Comment:
@enadeau this is TINY but could we change the name of this file to type-check or something to that effect?

Reply:
Done